### PR TITLE
Add support for golangci on ci

### PR DESCRIPTION
**Pull request by ffrizzo**
This pr is a complement of https://github.com/surrealdb/surrealdb.go/pull/24. 

Adding support to the [GolangCI](https://golangci-lint.run/) to add more code checks and validation to help with code quality.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  golangci:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '>=1.19.1'
```
Comment (EtienneBruines) [inline .github/workflows/lint.yml:22]: We might want/need to stick to the go 1.18 mentioned in the `go.mod` file.

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    - path: pkg/commands/run.go
+      text: "SA1019: e.cfg.Run.Deadline is deprecated: Deadline exists for historical compatibility and should not be used."
+
+    - path: pkg/golinters/gofumpt.go
+      text: "SA1019: settings.LangVersion is deprecated: use the global `run.go` instead."
+    - path: pkg/golinters/staticcheck_common.go
+      text: "SA1019: settings.GoVersion is deprecated: use the global `run.go` instead."
+    - path: pkg/lint/lintersdb/manager.go
+      text: "SA1019: (.+).(GoVersion|LangVersion) is deprecated: use the global `run.go` instead."
+
```
Comment (EtienneBruines) [inline .golangci.yml:110]: Are these excluded for a reason? Should they be removed from the exclude-list and the code be refactored in future commits?

File: db.go [hunk 3]
```diff
@@ -240,22 +234,16 @@ func (self *DB) resp(_ string, params []interface{}, res interface{}) (interface
...
 func isSlice(possibleSlice interface{}) bool {
 	slice := false
-
-	switch v := possibleSlice.(type) {
-	default:
-		res := fmt.Sprintf("%s", v)
-		if res == "[]" || res == "&[]" || res == "*[]" {
-			slice = true
-		}
+	res := fmt.Sprintf("%s", possibleSlice)
```
Comment (EtienneBruines) [inline db.go:244]: The original code was a `switch` over a `.(type)`, whilst yours is on the value.

**Review by EtienneBruines — CHANGES_REQUESTED** [review]
I'm not familiar enough with `golangci` (on GitHub) because I primarily use `GitLab`, but overall it seems to get the job done. Nice!

File: .golangci.yml (hunk 2 above)
Comment (ffrizzo) [reply .golangci.yml:110]: I will remove this because this is from another project.

**Comment by ffrizzo** [comment]
> I'm not familiar enough with `golangci` (on GitHub) because I primarily use `GitLab`, but overall it seems to get the job done. Nice!

`golangci` works in the same way on local or on Github Actions or GitlabCI

**Review by EtienneBruines — APPROVED** [review]

**Comment by EtienneBruines** [comment]
> golangci works in the same way on local or on Github Actions or GitlabCI

Now I have a new toy to try out, thank you! 

The PR LGTM. Hopefully someone can merge it soon. :+1: 

Awesome work @ffrizzo !

**Pull request merged**